modules/account/src/main/kotlin/edu/stanford/spezi/module/account/register/RegisterScreen.kt

```diff
@@ -164,41 +203,47 @@ fun RegisterScreen(
                     })
             }
         }
-
-        ValidatedOutlinedTextField(
-            modifier = Modifier.fillMaxWidth(),
-            value = uiState.selectedGender.value,
-            onValueChange = {
-                onAction(Action.TextFieldUpdate(it, TextFieldType.GENDER))
-            },
-            labelText = "Gender",
-            readOnly = true,
-            trailingIcon = {
-                IconButton(onClick = {
-                    onAction(Action.DropdownMenuExpandedUpdate(true))
-                }) {
-                    Icon(Icons.Filled.ArrowDropDown, contentDescription = "Select Gender")
-                }
-            },
-            errorText = uiState.selectedGender.error,
-        )
+        IconLeadingContent(
+            content = {
+                ValidatedOutlinedTextField(
+                    modifier = Modifier.fillMaxWidth(),
+                    value = uiState.selectedGender.value,
+                    onValueChange = {
+                        onAction(Action.TextFieldUpdate(it, TextFieldType.GENDER))
+                    },
+                    labelText = "Gender",
+                    readOnly = true,
+                    trailingIcon = {
+                        IconButton(onClick = {
+                            onAction(Action.DropdownMenuExpandedUpdate(true))
+                        }) {
+                            Icon(Icons.Filled.ArrowDropDown, contentDescription = "Select Gender")
+                        }
+                    },
+                    errorText = uiState.selectedGender.error,
+                )
+            })
 
         VerticalSpacer(height = Spacings.small)
         var isDatePickerDialogOpen by remember { mutableStateOf(false) }
-        ValidatedOutlinedTextField(
-            modifier = Modifier.fillMaxWidth(),
-            value = uiState.dateOfBirth?.format(DateTimeFormatter.ofPattern("dd MMMM yyyy")) ?: "",
-            onValueChange = { /* Do nothing as we handle the date through the DatePicker */ },
-            labelText = "Date of Birth",
-            singleLine = true,
-            keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Next),
-            trailingIcon = {
-                IconButton(onClick = { isDatePickerDialogOpen = true }) {
-                    Icon(Icons.Filled.Edit, contentDescription = "Select Date")
-                }
-            },
-            readOnly = true
-        )
+        IconLeadingContent(
+            content = {
+                ValidatedOutlinedTextField(
+                    modifier = Modifier.fillMaxWidth(),
+                    value = uiState.dateOfBirth?.format(DateTimeFormatter.ofPattern("dd MMMM yyyy"))
```

Comment:
Date formatting should ideally happen in view model when date is selected so that we also avoid multiple format calculations during compose recompositions. Alternatively you can use remember above `val dateString by remember`

Reply:
Good Idea!